capstone/capapi/views.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+from django.http import HttpResponse
+from django.db import IntegrityError
+from rest_framework import status, renderers, viewsets
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, list_route, renderer_classes
+from rest_framework.permissions import AllowAny
+from rest_framework.parsers import JSONParser, FormParser
+
+from . import models, serializers, resources, permissions
+from django.conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+class JSONResponse(HttpResponse):
+    """
+    An HttpResponse that renders its content into JSON.
+    """
+    def __init__(self, data, **kwargs):
+        content = renderers.JSONRenderer().render(data)
+        kwargs['content_type'] = 'application/json'
+        super(JSONResponse, self).__init__(content, **kwargs)
+
+
+class UserViewSet(viewsets.ModelViewSet):
+    serializer_class = serializers.UserSerializer
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    queryset = models.APIUser.objects.all()
+    parser_classes = (JSONParser, FormParser,)
+    lookup_field = 'email'
+
+    @list_route(methods=['get'], permission_classes=[AllowAny])
+    def register(self, request):
+        serializer = serializers.RegisterUserSerializer()
+
+        return Response({'serializer': serializer}, template_name='sign-up.html')
+
+    @list_route(methods=['post'], permission_classes=[AllowAny])
+    def register_user(self, request):
+        serializer = serializers.RegisterUserSerializer(data=request.data)
+        if serializer.is_valid():
+            try:
+                user = serializer.create({
+                    'email': request.data.get('email'),
+                    'password': request.data.get('password'),
+                    'first_name': request.data.get('first_name'),
+                    'last_name': request.data.get('last_name')
+                })
+                content = {
+                    'status': 'Success!',
+                    'message': 'Thank you. Please check your email %s for a verification link.' % user.email
+                }
+                return Response(content, template_name='sign-up-success.html')
+            except IntegrityError as e:
```

Comment:
Is this the flow for a duplicate email address? Does it make sense for it to use a different flow/template from other validation errors? If so maybe add a comment to explain ...